code/datums/components/rotation.dm

```diff
@@ -83,44 +76,47 @@
 	return COMPONENT_NOTRANSFER
 
 /datum/component/simple_rotation/UnregisterFromParent()
-	remove_verbs()
-	remove_signals()
+	RemoveVerbs()
+	RemoveSignals()
 	. = ..()
 
 /datum/component/simple_rotation/Destroy()
-	QDEL_NULL(can_user_rotate)
-	QDEL_NULL(can_be_rotated)
-	QDEL_NULL(after_rotation)
+	QDEL_NULL(CanUserRotate)
+	QDEL_NULL(CanBeRotated)
+	QDEL_NULL(AfterRotation)
 	//Signals + verbs removed via UnRegister
 	. = ..()
 
 /datum/component/simple_rotation/ClearFromParent()
-	remove_verbs()
+	RemoveVerbs()
 	return ..()
 
 /datum/component/simple_rotation/proc/ExamineMessage(datum/source, mob/user, list/examine_list)
 	SIGNAL_HANDLER
+	examine_list += span_notice("Alt + Right-click to rotate it clockwise. Alt + Left-click to rotate it counterclockwise.")
+	if(rotation_flags & ROTATION_REQUIRE_WRENCH)
+		examine_list += span_notice("This requires a wrench to be rotated.")
 
-	if(rotation_flags & ROTATION_ALTCLICK)
-		examine_list += span_notice("Alt-click to rotate it clockwise.")
-
-/datum/component/simple_rotation/proc/HandRot(datum/source, mob/user, rotation = default_rotation_direction)
+/datum/component/simple_rotation/proc/RotateRight(datum/source, mob/user)
 	SIGNAL_HANDLER
+	if(!istype(user))
+		return FALSE
+	Rotate(user, ROTATION_CLOCKWISE)
 
-	if(!can_be_rotated.Invoke(user, rotation) || !can_user_rotate.Invoke(user, rotation))
-		return
-	BaseRot(user, rotation)
-
-/datum/component/simple_rotation/proc/WrenchRot(datum/source, obj/item/I, mob/living/user)
+/datum/component/simple_rotation/proc/RotateLeft(datum/source, mob/user)
 	SIGNAL_HANDLER
+	if(!istype(user))
+		return FALSE
+	Rotate(user, ROTATION_COUNTERCLOCKWISE)
 
-	if(!can_be_rotated.Invoke(user,default_rotation_direction) || !can_user_rotate.Invoke(user,default_rotation_direction))
+/datum/component/simple_rotation/proc/Rotate(mob/user, rotation_type)
+	if(!istype(user))
```

Comment:
On these checks, do we realistically expect them to happen, or are they sanity checks?
If they are sanity checks then they should `stack_trace()` or `CRASH()` so that we know they happen.

Reply:
We consider `mob/user` to 100% always appear.  I made it so that `stack_trace` is called if we are missing it with https://github.com/tgstation/tgstation/pull/64441/commits/18ba5ba7f15b40352d5bfacae8dcb88383206ab3